systest/cluster/cluster.go

```diff
@@ -309,10 +309,12 @@ func (c *Cluster) AddSmeshers(cctx *testcontext.Context, n int) error {
 		return err
 	}
 	bootnodes := c.clients[:c.bootnodes]
+	smeshers := c.clients[c.bootnodes:]
 	c.clients = nil
+	smeshers = append(smeshers, clients...)
 	c.clients = append(c.clients, bootnodes...)
-	c.clients = append(c.clients, clients...)
-	c.smeshers = len(clients)
+	c.clients = append(c.clients, smeshers...)
+	c.smeshers = len(smeshers)
```

Comment:
i think less prone to bugs would be to avoid append on smeshers slice, e.g

```
c.clients = append(..., bootnodes...)
c.clients = append(..., smeshers...)
c.clients = append(...., clients...)
c.smeshers += len(clients)
```

Reply:
done